### PR TITLE
ADLS swagger updates

### PR DESCRIPTION
**Pull request by begoldsm**
1. Update to support account encryption
2. Re-add Expiry (get/set) for files
3. Re-add the ability to remove acls and default acls
4. Add tests for these things and run them.

### CODE REVIEW

**Comment by begoldsm** [comment]
@markcowl this is a much smaller PR :) Can we get this one in hopefully tonight? I suspect I am not going to be able to make code freeze for PowerShell, but I would like to try and at least make it into the release branch first thing in the morning.

File: src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs [hunk 1]
```diff
@@ -229,13 +321,19 @@ public void SecretAndCredentialCRUDTest()
                             credentialCreationScript);
 
                         // Get the Credential list
+                        // This funcitonality is removed in this release
```
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs]: remove commented code

File: src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs [hunk 2]
```diff
@@ -163,6 +163,98 @@ public void GetCatalogItemsTest()
...
+
+                    // Get the credential and ensure the response contains a date.
+                    var secretGetResponse = clientToUse.Catalog.GetCredential(
+                        commonData.SecondDataLakeAnalyticsAccountName,
+                        commonData.DatabaseName, commonData.SecretName);
+
+                    Assert.NotNull(secretGetResponse);
+                    Assert.NotNull(secretGetResponse.Name);
+
+                    // Get the Credential list
```
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs]: any reason commented code is still here?

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs [hunk 3]
```diff
@@ -1051,7 +1433,7 @@ internal CatalogOperations(DataLakeAnalyticsCatalogManagementClient client)
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<USqlExternalDataSource>> GetExternalDataSourceWithHttpMessagesAsync(string accountName, string databaseName, string externalDataSourceName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<USqlCredential>> GetCredentialWithHttpMessagesAsync(string accountName, string databaseName, string credentialName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs]: Isn't this a breaking change? Changing public surface area

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs [hunk 4]
```diff
@@ -1252,16 +1628,13 @@ internal CatalogOperations(DataLakeAnalyticsCatalogManagementClient client)
...
-        /// Thrown when unable to deserialize the response
-        /// </exception>
         /// <exception cref="ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<IPage<USqlExternalDataSource>>> ListExternalDataSourcesWithHttpMessagesAsync(string accountName, string databaseName, ODataQuery<USqlExternalDataSource> odataQuery = default(ODataQuery<USqlExternalDataSource>), string select = default(string), bool? count = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse> DeleteCredentialWithHttpMessagesAsync(string accountName, string databaseName, string credentialName, DataLakeAnalyticsCatalogCredentialDeleteParameters parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs]: breaking change?

File: src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs (hunk 1 above)
Comment (begoldsm) [inline src/ResourceManagement/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs]: This code is not there anymore :)

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs (hunk 4 above)
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs]: ignore this, looks like [https://github.com/Azure/azure-sdk-for-net/pull/2289/](https://github.com/Azure/azure-sdk-for-net/pull/2289/) already has the major/minor version updates.

---
In reply to: [77389407](https://github.com/Azure/azure-sdk-for-net/pull/2315#discussion_r77389407) [](ancestors = 77389407)

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs (hunk 3 above)
Comment (shahabhijeet) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs]: ignore this, looks like [https://github.com/Azure/azure-sdk-for-net/pull/2289/](https://github.com/Azure/azure-sdk-for-net/pull/2289/) already has the major/minor version updates.

---
In reply to: [77389322](https://github.com/Azure/azure-sdk-for-net/pull/2315#discussion_r77389322) [](ancestors = 77389322)

**Comment by shahabhijeet** [comment]
:shipit:

**Pull request merged**